src/random.ts

```diff
@@ -1,13 +1,13 @@
+import { range } from './standard'
+
 export const getRandom = (floor: number, ceiling: number): number => {
   return Math.floor(Math.random() * (ceiling - floor + 1)) + floor
 }
 
 export const getRandomWithExceptions = (floor: number, ceiling: number, exceptions: number[]): number => {
-  const random = getRandom(floor, ceiling)
-  if (exceptions.includes(random)) {
-    return getRandomWithExceptions(floor, ceiling, exceptions)
-  }
-  return random
+  const potentialResults = range(floor, ceiling)
+  const filteredResults = potentialResults.filter((result) => !exceptions.includes(result))
```

Comment:
I'm not entirely sure if this is optimal - we're avoiding the infinite loop, but we've increased the big O to ca. On, where n is the number of possibilities.
This is fine for the usecase we have for this function, but could cause issues in the future.

An alternative could be that we calculate the difference between floor and ceiling, and see if that is greater than the number of exceptions that are passed in?
Although not sure if that covers everything

Reply:
ahaaa, good idea!